src/Jenssegers/Mongodb/Eloquent/Model.php

```diff
@@ -233,11 +234,11 @@ public function getAttribute($key)
 
                 // This attribute matches an embedsOne or embedsMany relation so we need
                 // to return the relation results instead of the interal attributes.
-                if ($relations instanceof EmbedsOneOrMany) {
+                if ($relations instanceof EmbedsOneOrMany or $relations instanceof MorphTo) {
```

Comment:
Might be better to split the logic indeed, I'll have to check what exactly is going wrong because this is weird.